rest/utilities_testing.go

```diff
@@ -231,7 +230,39 @@ func (rt *RestTester) Bucket() base.Bucket {
 
 		rt.DatabaseConfig.SGReplicateEnabled = base.BoolPtr(rt.RestTesterConfig.sgReplicateEnabled)
 
-		_, err = rt.RestTesterServerContext.AddDatabaseFromConfig(*rt.DatabaseConfig)
+		if rt.RestTesterConfig.TestBucket != nil {
+			// If using a leaky bucket, make sure it ignores attempted closures to avoid double closures panic
+			// due to closing both RestTesterServerContext and testBucket in rt.Close()
+			leakyBucket, isLeaky := base.AsLeakyBucket(testBucket)
+			if isLeaky {
+				leakyBucket.SetIgnoreClose(true)
+			}
+
+			// Set scopes and collections on bucket if set in DB Config
+			var scope, collection *string
+			for scopeName, scopeConfig := range rt.RestTesterConfig.DatabaseConfig.Scopes {
+				scope = &scopeName
+				for collectionName := range scopeConfig.Collections {
+					collection = &collectionName
+					break
+				}
+			}
+			if scope != nil && collection != nil {
+				collectionBucket, isCollection := base.AsCollectionStore(rt.tb, testBucket.Bucket)
+				if !isCollection {
+					rt.tb.Fatalf("Could not get collection from bucket with type %T: %v", testBucket.Bucket, err)
+				}
+
+				collectionBucket.Spec.Scope = scope
+				collectionBucket.Spec.Collection = collection
+				collectionBucket.Collection = collectionBucket.Collection.Bucket().Scope(*scope).Collection(*collection)
+			}
+
+			_, err = rt.RestTesterServerContext.AddDatabaseFromConfigWithBucket(rt.tb, *rt.DatabaseConfig, testBucket.Bucket)
```

Comment:
should this be `collectionBucket` based on setting spec/collection above? Or doesn't it matter?

```suggestion
			_, err = rt.RestTesterServerContext.AddDatabaseFromConfigWithBucket(rt.tb, *rt.DatabaseConfig, collectionBucket)
```

Reply:
This would cause the leaky bucket to be lost and therefore not fire the callbacks